clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -1272,12 +1274,27 @@ private Meter createMeter(Metrics metrics, String groupName,  Map<String, String
             return createMeter(metrics, groupName, metricTags, null, baseName, descriptiveName);
         }
 
-        private Meter createIOThreadRatioMeter(Metrics metrics, String groupName,  Map<String, String> metricTags,
+        /**
+         * This method generates `time-total` metrics with a couple of errors, no `-ns` suffix and no dash between basename
+         * and `time-toal` suffix.
+         * @deprecated use {{@link #createIOThreadRatioMeter(Metrics, String, Map, String, String)}} for new metrics instead
+         */
+        @Deprecated
+        private Meter createIOThreadRatioMeterLegacy(Metrics metrics, String groupName,  Map<String, String> metricTags,
```

Comment:
Keeping the old method for compatibility reasons, but added a javadoc and deprecation to it